server/src/modules/task/task.controller.ts

```diff
@@ -44,25 +44,40 @@ export class TaskController {
   async createTask(
     @Request() req: RequestType,
     @Body() taskData: Task,
-  ): Promise<Task> {
+  ): Promise<TaskInterface[]> {
     const userId = req.user?.id;
     if (userId) {
-      return this.taskService.createTask(taskData, userId);
+      await this.taskService.createTask(taskData, userId);
+      return this.taskService.getAllTasks(userId);
     } else {
       throw new UnauthorizedException();
     }
   }
 
   @Put(':id')
   async updateTask(
+    @Request() req: RequestType,
     @Param('id') id: number,
     @Body() taskData: Task,
-  ): Promise<Task> {
-    return this.taskService.updateTask(id, taskData);
+  ): Promise<TaskInterface[]> {
+    if (req.user) {
```

Comment:
nit: for this type of code, I prefer
```
if (!req.user) {
  throw new UnauthorizedException();
}

await this.taskService.updateTask(id, taskData);
return this.taskService.getAllTasks(req.user.id);
```

It's the same idea as avoiding using nested `.then` statements: you want the primary logic of your code to flow nicely. If there's a problem, you can throw an exception or return early, and then the rest of your code doesn't have extra braces/indents that break it up.